### ISSUE TITLE
[Slider] Discrete Slider increment/decrement should match discreteValues

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#3040]
### Overview
When numberOfDiscreteValues is non-zero, the MDCSlider should use these as the step value in `accessibilityIncrement` and `accessibilityDecrement`.  If the number of discrete values is fewer than 8, then VoiceOver users cannot modify the slider value by swiping up or down.

### Reproduction steps
```objectivec
- (void)testAccessibilityIncrementDiscreteSlider {
  for (NSUInteger i = 2; i < 20; ++i) {
    // Given
    self.slider.numberOfDiscreteValues = i;
    self.slider.minimumValue = 0;
    self.slider.maximumValue = 1;
    self.slider.value = self.slider.minimumValue;

    // When
    [self.slider accessibilityIncrement];

    // Then
    CGFloat stepValue = (self.slider.maximumValue - self.slider.minimumValue) /
        (self.slider.numberOfDiscreteValues - 1);
    CGFloat expectedValue = self.slider.minimumValue + stepValue;
    XCTAssertEqualWithAccuracy(self.slider.value, expectedValue, 0.0001,
                               @"A slider with (%lu) discrete values should have a step value of '%.3f'.",
                               (unsigned long)self.slider.numberOfDiscreteValues, stepValue);
  }
}
```

### Version number
46.0.0
